kamon-core/src/main/scala/kamon/trace/Sampler.scala

```diff
@@ -71,3 +71,16 @@ class ThresholdSampler(thresholdInNanoseconds: Long) extends Sampler {
   def shouldReport(traceElapsedTime: NanoInterval): Boolean = traceElapsedTime.nanos >= thresholdInNanoseconds
 }
 
+class ClockSampler(pauseInNanoseconds: Long) extends Sampler {
```

Comment:
The parameter should be also a `NanoInterval` to keep it consistent across samplers. Also, the current implementation is not thread-safe because of the `timer` var bellow which needs to be safely published and coordinated. This code is going to be called from multiple threads so some changes need to be made.